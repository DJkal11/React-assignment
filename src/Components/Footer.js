

function Footer() {
  return (
    
<footer data-testid="footer" className="p-6 bg-gray-200 text-center">
    <span className="text-sm m-auto text-gray-500">© 2022 <a href="/" className="hover:underline">XYZ™</a>. All Rights Reserved.
    </span>
    
</footer>

  );
}

export default Footer;