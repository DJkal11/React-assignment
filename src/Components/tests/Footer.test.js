import { render, screen, cleanup } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";

import Footer from "../Footer"

afterEach(() => {
  cleanup()
})

test("should render footer component", () => {
    render(<Footer/>)
    const headerComponent = screen.getByTestId('footer');
    expect(headerComponent).toBeInTheDocument();
})

test("Footer includes correct html innerText", () => {
    render(<Footer/>)
    const headerComponent = screen.getByTestId('footer');
    expect(headerComponent).toHaveTextContent(". All Rights Reserved.");
})

test("renders Footer snapshot", () => {
    const newFooter = renderer.create(<Footer/>).toJSON();
    expect(newFooter).toMatchSnapshot();
})