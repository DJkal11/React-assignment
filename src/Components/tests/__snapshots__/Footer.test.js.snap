// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer snapshot 1`] = `
<footer
  className="p-6 bg-gray-200 text-center"
  data-testid="footer"
>
  <span
    className="text-sm m-auto text-gray-500"
  >
    © 2022 
    <a
      className="hover:underline"
      href="/"
    >
      XYZ™
    </a>
    . All Rights Reserved.
  </span>
</footer>
`;
